operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java

```diff
@@ -420,6 +424,21 @@ private void addEnvVars(StatefulSet baseDeployment, Keycloak keycloakCR, TreeSet
         // include the kube CA if the user is not controlling KC_TRUSTSTORE_PATHS via the unsupported or the additional
         varMap.putIfAbsent(KC_TRUSTSTORE_PATHS, new EnvVarBuilder().withName(KC_TRUSTSTORE_PATHS).withValue(truststores).build());
 
+        varMap.putIfAbsent(KC_TRACING_SERVICE_NAME,
+                new EnvVarBuilder().withName(KC_TRACING_SERVICE_NAME)
+                        .withValue(keycloakCR.getMetadata().getName())
+                        .build()
+        );
+
+        // Possible OTel k8s attributes convention can be found here: https://opentelemetry.io/docs/specs/semconv/attributes-registry/k8s/#kubernetes-attributes
+        var tracingAttributes = Map.of("k8s.namespace.name", keycloakCR.getMetadata().getNamespace());
+
+        varMap.putIfAbsent(KC_TRACING_RESOURCE_ATTRIBUTES,
+                new EnvVarBuilder().withName(KC_TRACING_RESOURCE_ATTRIBUTES)
+                        .withValue(tracingAttributes.entrySet().stream().map(e -> e.getKey() + "=" + e.getValue()).collect(Collectors.joining(",")))
```

Comment:
Not a problem for what's here, but just wanted to double check something. Looking at the well known key / values if you wanted to add annotations those values can contain commas, which would likely throw off the parsing. Is there escaping for commas, or would attributes like that not be supported?

Reply:
@shawkins I'm not sure if I understand. Could you elaborate a little bit more?

Are you referring to the resource attributes specified in the env var and then parsing on the server side? The parsing of these comma-separated values should be covered in this test case: https://github.com/keycloak/keycloak/blob/main/quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/TracingDistTest.java#L176

Reply:
There will be currently always only a single entry `k8s.namespace.name={namespace}` if I understand correctly. So no need to handle escaping the commas?

Reply:
@vmuzikar correct, the namespace value will not contain , nor \

Reply:
@shawkins Good point. TBH, I haven't further investigated such a "corner" case and rather have been relying on the Quarkus handling as these attributes are a list of items in the format `key1=val1,key2=val2,key3=val3`[1]. 

I think it's sufficient ATM.

> There will be currently always only a single entry k8s.namespace.name={namespace} if I understand correctly. So no need to handle escaping the commas?

For the future, we might consider adding other attributes besides the namespace. 

[1] https://quarkus.io/guides/opentelemetry#quarkus-opentelemetry_quarkus-otel-resource-attributes